src/transformStateWithClones.js

```diff
@@ -7,7 +7,36 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const dataOfState = [];
+  const stateCopy = { ...state };
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(stateCopy, action.extraData);
+        dataOfState.push({ ...stateCopy });
```

Comment:
you can use `dataOfState.push({ ...stateCopy })` just once, and write it between lines 36 and 37